src/main/kotlin/uk/gov/dluhc/printapi/mapper/PrintDetailsMapper.kt

```diff
@@ -0,0 +1,30 @@
+package uk.gov.dluhc.printapi.mapper
+
+import org.apache.commons.lang3.RandomStringUtils
+import org.bson.types.ObjectId
+import org.mapstruct.Mapper
+import org.mapstruct.Mapping
+import uk.gov.dluhc.printapi.database.entity.PrintDetails
+import uk.gov.dluhc.printapi.dto.EroManagementApiEroDto
+import uk.gov.dluhc.printapi.messaging.models.CertificateLanguage
+import uk.gov.dluhc.printapi.messaging.models.SendApplicationToPrintMessage
+import java.util.UUID
+
+@Mapper(uses = [SourceTypeMapper::class], imports = [UUID::class, ObjectId::class, RandomStringUtils::class])
+abstract class PrintDetailsMapper {
+    @Mapping(target = "id", expression = "java(UUID.randomUUID())")
+    @Mapping(target = "requestId", expression = "java( new ObjectId().toHexString())")
```

Comment:
For these 2 (`requestId` and `vacNumber`), are we not going to use the `IdFactory` we have?
We will 100% need to do that once we know the final format for the VAC Number - it's not going to be a simple algorithm and will likely need the gssCode, date, and (possibly?) previous VAC number if there was one (but ignore that last part for now)

Can you have a look at using the `IdFactory` please?

Reply:
Hi @NathanRussellValtech ,

Thanks, for the suggestion. 

When I was implementing the mapper, I actually forgot we had the `IdFactory` class. 
I remembered it after pushing the changes and wanted to have your thoughts on the above implementation.
I agree with you. It makes sense to use the factory class, if the generation algorithm will be more complicated in the future.

I will change the code accordingly.

Reply:
Maybe having the expressions in separate referenced functions (and not use expression at all) so that the mapper is kept simple and then there can be specific tests for these which would simplify the test class.